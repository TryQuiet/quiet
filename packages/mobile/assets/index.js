import zbay_icon from './icons/zbay_icon.png';
import quiet_icon from './icons/quiet_icon.png';
import icon_send from './icons/icon_send.png';
import icon_send_disabled from './icons/icon_send_disabled.png';
import check_circle_green from './icons/check_circle_green.png';
import check_circle_blank from './icons/check_circle_blank.png';
import username_registered from './icons/username_registered.png';
import arrow_left from './icons/arrow_left.png';
import icon_warning from './icons/icon_warning.png';

/**
 * @description This assets are for the app.
 */
export const appImages = {
  zbay_icon,
  quiet_icon,
  icon_send,
  icon_send_disabled,
  check_circle_green,
  check_circle_blank,
  username_registered,
  arrow_left,
  icon_warning
};

/**
 * @description This assets are for the storybook.
 */
export const storybookImages = {};
