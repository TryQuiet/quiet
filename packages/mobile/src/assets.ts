import arrow_left from '../assets/icons/arrow_left.png'
import arrow_right_short from '../assets/icons/arrow_right_short.png'
import check_circle_blank from '../assets/icons/check_circle_blank.png'
import check_circle_green from '../assets/icons/check_circle_green.png'
import dots from '../assets/icons/dots.png'
import paperclip_gray from '../assets/icons/paperclip_gray.png'
import file_document from '../assets/icons/file_document.png'
import icon_check_white from '../assets/icons/icon_check_white.png'
import icon_close from '../assets/icons/icon_close.png'
import icon_send from '../assets/icons/icon_send.png'
import icon_send_disabled from '../assets/icons/icon_send_disabled.png'
import icon_warning from '../assets/icons/icon_warning.png'
import quiet_icon from '../assets/icons/quiet_icon.png'
import quiet_icon_round from '../assets/icons/quiet_icon_round.png'
import update_graphics from '../assets/icons/update_graphics.png'
import username_registered from '../assets/icons/username_registered.png'

/**
 * @description This assets are for the app.
 */
export const appImages = {
  arrow_left,
  arrow_right_short,
  check_circle_blank,
  check_circle_green,
  dots,
  paperclip_gray,
  file_document,
  icon_check_white,
  icon_close,
  icon_send,
  icon_send_disabled,
  icon_warning,
  quiet_icon,
  quiet_icon_round,
  update_graphics,
  username_registered,
}

/**
 * @description This assets are for the storybook.
 */
export const storybookImages = {}
