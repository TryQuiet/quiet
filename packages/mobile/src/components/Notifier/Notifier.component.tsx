import React, { FC } from 'react'
import { View, Text, Image } from 'react-native'

import { Typography } from '../Typography/Typography.component'
import { Button } from '../Button/Button.component'

import { NotifierProps } from './Notifier.types'

import { defaultTheme } from '../../styles/themes/default.theme'

export const Notifier: FC<NotifierProps> = ({ onButtonPress, onEmailPress, icon, title, message, style }) => {
  return (
    <View
      style={{
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: defaultTheme.palette.background.white,
      }}
    >
      <Image
        source={icon}
        style={{
          margin: 20,
          resizeMode: 'cover',
          width: 375,
          height: 135,
        }}
      />
      <View style={{ flexDirection: 'column', alignItems: 'center', gap: 20, maxWidth: 330 }}>
        <Typography fontSize={28} fontWeight={'medium'} horizontalTextAlign={'center'}>
          {title}
        </Typography>
        <Typography fontSize={14} fontWeight={'normal'} horizontalTextAlign={'center'}>
          {message}
        </Typography>
        <Button onPress={onButtonPress} title={'I understand'} width={135} newDesign />
        <Typography onPress={onEmailPress} fontSize={16} color={'gray50'}>
          Need help? <Text style={{ textDecorationLine: 'underline' }}>help@quiet.chat</Text>
        </Typography>
      </View>
    </View>
  )
}
