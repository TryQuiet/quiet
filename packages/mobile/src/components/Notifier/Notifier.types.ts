import { TextStyle } from 'react-native'

export interface NotifierProps {
  onButtonPress: () => void
  onEmailPress: () => void
  icon: any
  title: string
  message: string
  style?: TextStyle
}
