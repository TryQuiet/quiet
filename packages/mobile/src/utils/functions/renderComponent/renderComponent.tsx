import { render, RenderAPI } from '@testing-library/react-native'
import React, { FC, ReactElement } from 'react'
import { ThemeProvider } from 'styled-components/native'

import { defaultTheme } from '../../../styles/themes/default.theme'
import { RenderComponentProps } from './renderComponent.types'

export const renderComponent = (ui: ReactElement): RenderAPI => {
  const Wrapper: FC<RenderComponentProps> = ({ children }) => (
    <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
  )

  return render(ui, { wrapper: Wrapper })
}
