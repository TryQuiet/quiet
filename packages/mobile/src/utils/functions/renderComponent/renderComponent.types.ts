export interface RenderComponentProps {
  children?: React.ReactNode
}
