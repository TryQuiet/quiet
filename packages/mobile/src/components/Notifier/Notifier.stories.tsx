import React from 'react'
import { storiesOf } from '@storybook/react-native'

import { appImages } from '../../assets'
import { storybookLog } from '../../utils/functions/storybookLog/storybookLog.function'

import { Notifier } from './Notifier.component'

storiesOf('Notifier', module).add('Default', () => (
  <Notifier
    onButtonPress={storybookLog('button pressed')}
    onEmailPress={storybookLog('email pressed')}
    icon={appImages.update_graphics}
    title={'Coming update will remove communities & messages'}
    message={
      'Quietâ€™s next release makes joining communities faster and more reliable by letting people join when the owner is offline! ðŸŽ‰ However, these changes required us to reset all communities, and both communities and messages will be lost on mobile. ðŸ˜¥ We apologize for the inconvenience, and please reach out immediately if you need help backing up messages.'
    }
  />
))
