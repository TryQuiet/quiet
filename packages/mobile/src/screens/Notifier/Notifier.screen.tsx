import React, { FC, useCallback } from 'react'
import { Linking } from 'react-native'
import { Notifier } from '../../components/Notifier/Notifier.component'
import { appImages } from '../../assets'
import { useDispatch } from 'react-redux'
import { navigationActions } from '../../store/navigation/navigation.slice'

export const NotifierScreen: FC = () => {
  const dispatch = useDispatch()

  const redirection = useCallback(() => {
    dispatch(navigationActions.pop())
  }, [dispatch])

  const helpline = useCallback(async () => {
    const link = 'mailto:help@quiet.chat'
    await Linking.openURL(link);
  }, [])

  return (
    <Notifier
      onButtonPress={redirection}
      onEmailPress={helpline}
      icon={appImages.update_graphics}
      title={'Coming update will remove communities & messages'}
      message={
        'Quietâ€™s next release makes joining communities faster and more reliable by letting people join when the owner is offline! However, these changes required us to reset all communities, and both communities and messages will be lost on mobile. We apologize for the inconvenience, and please reach out immediately if you need help backing up messages.'
      }
    />
  )
}
