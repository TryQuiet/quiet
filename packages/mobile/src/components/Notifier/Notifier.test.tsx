import React from 'react'
import '@testing-library/jest-native/extend-expect'
import { fireEvent, screen } from '@testing-library/react-native'

import { renderComponent } from '../../utils/functions/renderComponent/renderComponent'

import { appImages } from '../../assets'

import { Notifier } from './Notifier.component'

describe('Notifier component', () => {
  it('should match inline snapshot', () => {
    const { toJSON } = renderComponent(
      <Notifier
        onButtonPress={jest.fn()}
        onEmailPress={jest.fn()}
        icon={appImages.update_graphics}
        title={'Coming update will remove communities & messages'}
        message={
          'Quietâ€™s next release makes joining communities faster and more reliable by letting people join when the owner is offline! ðŸŽ‰ However, these changes required us to reset all communities, and both communities and messages will be lost on mobile. ðŸ˜¥ We apologize for the inconvenience, and please reach out immediately if you need help backing up messages.'
        }
      />
    )

    expect(toJSON()).toMatchSnapshot()
  })

  it('should respond on button tap', () => {
    const buttonCallback = jest.fn()

    renderComponent(
      <Notifier
        onButtonPress={buttonCallback}
        onEmailPress={jest.fn()}
        icon={appImages.update_graphics}
        title={'Coming update will remove communities & messages'}
        message={
          'Quietâ€™s next release makes joining communities faster and more reliable by letting people join when the owner is offline! ðŸŽ‰ However, these changes required us to reset all communities, and both communities and messages will be lost on mobile. ðŸ˜¥ We apologize for the inconvenience, and please reach out immediately if you need help backing up messages.'
        }
      />
    )

    fireEvent.press(screen.getByText('I understand'))

    expect(buttonCallback).toBeCalled()
  })
})
