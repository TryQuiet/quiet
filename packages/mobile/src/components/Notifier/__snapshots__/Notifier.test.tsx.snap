// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifier component should match inline snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../assets/icons/update_graphics.png",
      }
    }
    style={
      {
        "height": 135,
        "margin": 20,
        "resizeMode": "cover",
        "width": 375,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "column",
        "gap": 20,
        "maxWidth": 330,
      }
    }
  >
    <Text
      color="main"
      fontSize={28}
      fontWeight="medium"
      horizontalTextAlign="center"
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "Rubik-Medium",
            "fontSize": 28,
            "textAlign": "center",
            "textAlignVertical": "center",
          },
        ]
      }
      verticalTextAlign="center"
    >
      Coming update will remove communities & messages
    </Text>
    <Text
      color="main"
      fontSize={14}
      fontWeight="normal"
      horizontalTextAlign="center"
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "Rubik-Regular",
            "fontSize": 14,
            "textAlign": "center",
            "textAlignVertical": "center",
          },
        ]
      }
      verticalTextAlign="center"
    >
      Quietâ€™s next release makes joining communities faster and more reliable by letting people join when the owner is offline! However, these changes required us to reset all communities, and both communities and messages will be lost on mobile. We apologize for the inconvenience, and please reach out immediately if you need help backing up messages.
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#521C74",
          "borderRadius": 16,
          "justifyContent": "center",
          "marginVertical": 12,
          "minHeight": 50,
          "paddingVertical": 12,
          "width": 135,
        }
      }
      testID="button"
    >
      <Text
        color="white"
        fontSize={16}
        horizontalTextAlign="left"
        style={
          [
            {
              "color": "#ffffff",
              "fontFamily": "Rubik-Regular",
              "fontSize": 16,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
          ]
        }
        verticalTextAlign="center"
      >
        I understand
      </Text>
    </View>
    <Text
      color="gray50"
      fontSize={16}
      horizontalTextAlign="left"
      onPress={[MockFunction]}
      style={
        [
          {
            "color": "#7F7F7F",
            "fontFamily": "Rubik-Regular",
            "fontSize": 16,
            "textAlign": "left",
            "textAlignVertical": "center",
          },
        ]
      }
      verticalTextAlign="center"
    >
      Need help? 
      <Text
        style={
          {
            "textDecorationLine": "underline",
          }
        }
      >
        help@quiet.chat
      </Text>
    </Text>
  </View>
</View>
`;
